internal/database/mock.go

```diff
@@ -0,0 +1,33 @@
+package database
+
+import (
+	"context"
+
+	"github.com/alexdunne/gs-onboarding/internal/models"
+	"github.com/stretchr/testify/mock"
+)
+
+type Mock struct {
+	mock.Mock
+}
+
+func (m *Mock) GetAll(ctx context.Context) ([]models.Item, error) {
+	args := m.Called(ctx)
+	return args.Get(0).([]models.Item), args.Error(1)
+}
+
+func (m *Mock) GetStories(ctx context.Context) ([]models.Item, error) {
+	args := m.Called(ctx)
+	return args.Get(0).([]models.Item), args.Error(1)
```

Comment:
you might want to check if the value is ok on casting

Reply:
Q:

If I change it something like:

```
func (m *Mock) FetchAll(ctx context.Context) ([]models.Item, error) {
	args := m.Called(ctx)

	itemsArg, ok := args.Get(0).([]models.Item)
	if !ok {
		// ___ HERE ___
	}

	return itemsArg, args.Error(1)
}
```

What should I do in the case that it doesn't match what we expect. Initially I was going to return an `errInvalidType` or something but that could mean some tests incorrectly pass if they were expecting to receive an error. In this case should I `panic` because it's a mock or is there a better approach?

Reply:
We tend to return an empty version of what was expected e.g. empty list or nil etc
Then to your other point tests won't pass as they won't get the expected data